### DISCUSSION TITLE
does next12 drops support of toplevelawait ?

### DISCUSSION THREAD

**Discussion by PookMook** [vercel/next.js#30512] [Help]
It seems nextjs 12 drops the support of toplevelawait workaround discussed here : https://github.com/vercel/next.js/discussions/11185
it's not really babel related so it didn't cut it for https://github.com/vercel/next.js/discussions/30174

I updated my app yesterday and can't get swc to accept toplevelawait like webpack5 did. I can see some people mentioning it on SWC github but not sure how to proceed from there. Adding a .swcrc doesn't do anything sadly

PS: reading releases looks like 12.0.1 disables `.swcrc`, going back to a 12.0.0 actually builds with webpack when a .swcrc is present
It looks like I just have to give the team some time to workout the kinks of a major release before they re-enable swc customization

**Comment by karlhorky** [comment]
cc @Janpot should the `topLevelAwait` experiment be gone in Next.js 12?

Aside from this discussion, it seems other are also having issues with it: https://github.com/vercel/next.js/pull/30082#issuecomment-952673352

**Reply by Janpot** [reply]
Looks like a [`.babelrc` was recently added](https://github.com/vercel/next.js/pull/29810/) to the [TLA test](https://github.com/vercel/next.js/tree/canary/test/integration/async-modules) to make it opt it out of using the swc loader. I guess it's still supported, but you'll need to use the babel loader for now.

**Reply by sajadghawami** [reply]
@Janpot 

could you clearify on this? How would can we add this?

**Reply by Janpot** [reply]
If you really want to use top level await right now, you could opt-out of the use of `swc` as the JS compiler by adding to your project a `.babelrc` containing
```json
{
  "presets": ["next/babel"]
}
```
Here's an example of a [next.js app that supports TLA](https://github.com/vercel/next.js/tree/canary/test/integration/async-modules). Be warned that you'll take a massive performance hit by doing this.

**Reply by karlhorky** [reply]
I guess it's planned to support TLA with `swc` in the future too? Is there already a tracking issue in either the Vercel repos or swc itself?

Edit: Ah I guess swc itself already supports top-level await https://github.com/swc-project/swc/commit/94eac1de89a69ce0f2b30ced7d1c3f9fe48640aa

**Reply by sajadghawami** [reply]
nice, thanks!

if `swc` already supports TLA, cant we just pass a flag to it or am i missing something?

**Reply by Janpot** [reply]
maybe, feel free to look into it.

**Reply by sajadghawami** [reply]
> Loading `.swcrc` has been disabled in 12.0.1 as it was unintentionally enabled and causes issues like these where customizations don't work correctly yet. Please post your custom Babel configuration https://github.com/vercel/next.js/discussions/30174 as it helps with prioritization.

_Originally posted by @timneutkens in https://github.com/vercel/next.js/discussions/30413#discussioncomment-1543513_


gotta wait then :)

**Reply by timneutkens** [reply]
Created #31054 to track this 🙏

**Reply by PookMook** [reply]
Thanks Tim, greatly appreciated!
